### DISCUSSION TITLE
（help）Hilight Output line less than Input

### DISCUSSION THREAD

**Discussion by vanker-x** [pygments/pygments#2664] [Q&A] [answered]
Hello everyone, I just came into contact with pygments recently, and then I had some problems in using it. When I used pygments to highlight my exception stack code, I found that the html code output after highlighting would be less than the number of lines of code I originally input and I can be sure that the number of lines of my code is fixed at 10 lines. Can anyone tell me why and im all ears 
The following is my environment information and sample code.

Env:
    Pygments: 2.17.2
    Python: 3.12.2
    OS: MacOS ventura 13.6.4

```python
my_input = """


def raise_error(count=3):
    if count == 0:
        raise ValueError()
    raise_error(count - 1)


def main():
    try:
"""
with open("error_trace.html", "w") as f:
    code = highlight(my_input, PythonLexer(),
                     HtmlFormatter(filename="foo", linenos=True, linenostart=149,
                                   hl_lines=[6], full=True,style='monokai'))
    f.write(code)

```

my output
<img width="384" alt="QQ20240315-175055@2x" src="https://github.com/pygments/pygments/assets/59191502/9039fa48-3498-4b2b-82d9-af26afdeca30">

**Comment by jeanas** [comment, marked as answer]
By default, Pygments removes newlines at the beginning and at the end. You can override this by constructing the lexer as [`PythonLexer(stripnl=False)`](https://pygments.org/docs/lexers/).

**Reply by vanker-x** [reply]
Thank you for your wonderful reply
